nagios/datadog_checks/nagios/nagios.py

```diff
@@ -222,78 +196,46 @@ def check(self, instance):
 
-        if file_template is not None:
-            self.compile_file_template(file_template)
-
-        self.tail = TailFile(self.log, self.log_path, self._parse_line)
-        self.gen = self.tail.tail(line_by_line=False, move_end=True)
+        tail = TailFile(self.log, self.log_path, parse_line)
+        self.gen = tail.tail(line_by_line=False, move_end=True)
         next(self.gen)
 
     def check(self):
         self._line_parsed = 0
         # read until the end of file
         try:
-            self.log.debug("Start nagios check for file %s" % self.log_path)
+            self.log.debug("Start nagios check for file %s", self.log_path)
             next(self.gen)
-            self.log.debug("Done nagios check for file %s (parsed %s line(s))" % (self.log_path, self._line_parsed))
+            self._line_parsed += 1
+            self.log.debug("Done nagios check for file %s (parsed %s line(s))", self.log_path, self._line_parsed)
         except StopIteration as e:
             self.log.exception(e)
-            self.log.warning("Can't tail %s file" % self.log_path)
+            self.log.warning("Can't tail %s file", self.log_path)
 
-    def compile_file_template(self, file_template):
-        try:
-            # Escape characters that will be interpreted as regex bits
-            # e.g. [ and ] in "[SERVICEPERFDATA]"
-            regex = re.sub(r'[\[\]*]', r'.', file_template)
-            regex = re.sub(r'\$([^\$]*)\$', r'(?P<\1>[^\$]*)', regex)
-            self.line_pattern = re.compile(regex)
-        except Exception as e:
-            raise InvalidDataTemplate("%s (%s)" % (file_template, e))
 
-
-class NagiosEventLogTailer(NagiosTailer):
-    def __init__(
-        self, log_path, file_template, logger, hostname, tags, event_func, gauge_func, freq, passive_checks=False
-    ):
+class NagiosEventLogTailer(object):
+    def __init__(self, log_path, logger, hostname, event_func, tags, passive_checks):
         """
         :param log_path: string, path to the file to parse
-        :param file_template: string, format of the perfdata file
         :param logger: Logger object
         :param hostname: string, name of the host this agent is running on
         :param event_func: function to create event, should accept dict
-        :param gauge_func: function to report a gauge
-        :param freq: int, size of bucket to aggregate perfdata metrics
         :param passive_checks: bool, enable or not passive checks events
         """
-
-        self.passive_checks = passive_checks
-        self.tags = tags
-        super(NagiosEventLogTailer, self).__init__(
-            log_path, file_template, logger, hostname, event_func, gauge_func, freq
-        )
+        self.log = logger
+        self.hostname = hostname
+        self._event = event_func
+        self._tags = tags
+        self._passive_checks = passive_checks
+        self.check = NagiosTailer(log_path, logger, self._parse_line).check
```

Comment:
Should we keep an explicit reference to the tailer created here?

Reply:
What for?

Reply:
Nothing in mind currently